### PR TITLE
fix(ui): drop the slash-command echo of the local user message in the ACP adapter

### LINKED ISSUE #11670: Desktop shows a slash-command message twice (agent echoes the prompt into the ACP stream)
## Summary

In the desktop app, sending any slash command (`/plan ...`, `/goal ...`, `/summarize`, ...) shows the user's message twice in the thread.

- On builds before #10716 the two copies are glued into one user bubble with no separator, e.g. `/plan add tests/plan add tests`.
- On current `main` (after #10716) the echo carries a server-generated id, so it renders as a second, identical user bubble directly under the first.

Plain (non-slash) messages are not affected.

## Root cause

1. Every slash-command branch in `Agent::reply_impl` yields the user's own prompt back into the reply stream: `command_preamble = vec![AgentEvent::Message(user_message.clone()), ...]` and `yield AgentEvent::Message(user_message)` in `crates/goose/src/agents/agent.rs`. The normal prompt path never does this.
2. The ACP server forwards that message as a `user_message_chunk` (`handle_message_content` in `crates/goose/src/acp/server.rs`). Since #10716 the chunk carries the generated message id; before that it had none.
3. The desktop already appended the typed message to the store with a local id before calling `session/prompt` (`useChatSession.ts`). In `applyContentChunk` (`ui/desktop/src/acp/adapter/messages.ts`):
   - an id-less user chunk resolves to `lastMergeableMessageWithRole(state, 'user')`, i.e. the local message, and `lastContent.text += content.text` doubles the text;
   - a chunk with an unknown id is not adopted by the local message (it already has an id), so a new user message is pushed.

## Proposed fix

Client side, in `applyContentChunk`: ignore a non-steer `user_message_chunk` whose text equals the single text block of the last locally rendered user message. This covers both the id-less and the server-id shapes, leaves steer confirmation untouched (steer chunks are reconciled by id and replace text rather than append), and keeps the agent stream unchanged for the CLI and other ACP clients.

Removing the echo on the agent side would also fix it but changes what the CLI and third-party ACP clients receive, so it seems better handled separately (see #9261 for the structured-lifecycle direction).

## Verification plan

- Unit tests in `sessionNotificationAdapter.test.ts`: the echo is dropped in both shapes, a following agent chunk still lands as a separate message, and a genuine id-less user delta still merges.
- Manual: send `/plan add tests` in the desktop; the command appears once.

### PR DESCRIPTION
**Pull request by yixuanzhong**
Closes #11670

## Summary

Reworked to the server-side approach requested in review: the echo now stops in the ACP server, not in the desktop client, so every ACP client is fixed and the guard no longer depends on the prompt text.

Slash-command branches of `Agent::reply` yield the user's own prompt back into the reply stream (`command_preamble` / `yield AgentEvent::Message(user_message)`), which the plain prompt path never does. The server forwarded that echo as a `user_message_chunk`, so the client showed the prompt twice: glued into the local bubble before #10716, and as a second identical bubble now that the chunk carries a message id.

- `on_prompt` tags the prompt message `prompt_{uuid}`, the way `on_steer_session` already tags its own message with `steer_{uuid}`. `Message::with_generated_id_if_missing` leaves an id that is already set alone, so the tag reaches the echo unchanged.
- `handle_message_content` drops content of a user message carrying that id. It is the one point where agent messages leave the server, so this covers all ten echo sites at once.
- The id is threaded through `forward_agent_stream`; the `session/load` resume path passes `None` and is unaffected.

Matching on the id rather than the text also survives the `sanitize_unicode_tags` rewrite that `convert_acp_prompt_to_message` applies to the prompt, which the previous text-equality version missed.

Steer confirmations still reach the client: they carry their own `steer_` id. Session history is unchanged -- the prompt is still persisted, so replay after `session/load` shows it once.

### Testing

- New unit tests in `crates/goose/src/acp/server.rs`: the echo is suppressed by id, it is still suppressed when the prompt text was rewritten, steer and assistant messages with other ids are untouched, an id-less user message is untouched when no prompt id is tracked, and the assigned id survives the agent's id backfill.
- `cargo test -p goose --lib acp::server::tests`: 52 passed, 0 failed. `cargo fmt --check` clean.

### Related Issues

Discussion: #9261 (structured slash-command lifecycle events would remove the need for the echo altogether)

### CODE REVIEW

File: crates/goose/src/acp/server.rs [hunk 1]
```diff
@@ -3694,4 +3721,55 @@ print(\"hello, world\")
...
         client.await.unwrap().unwrap();
     }
+
+    #[test]
+    fn echoed_prompt_message_is_suppressed() {
+        let message = Message::user()
+            .with_text("/plan add tests")
+            .with_id("prompt_1");
+
+        assert!(is_echoed_prompt(&message, Some("prompt_1")));
```
Comment (alexhancock) [inline crates/goose/src/acp/server.rs:3731]: This test tests that `is_echoed_prompt` returns true when the id matches, but not really anything else. Likewise `steer_message_is_not_suppressed` is similarly not doing much on the opposite side.

If we're going to include tests it feels important to test the more meaningful bits about what happens to the conversation state, the UX, etc.

Comment (yixuanzhong) [reply crates/goose/src/acp/server.rs]: Fair point. Replaced them with an end-to-end test in `acp_server_test.rs` (`test_slash_command_does_not_echo_the_prompt_to_the_client`, 9401f6456) that drives `/skills` through the real ACP server and asserts the observable behaviour:

- the turn streams no `user_message_chunk` back to the client (the client already rendered the prompt),
- the command's reply still arrives as an `agent_message_chunk`,
- `load_session` replays the prompt exactly once, followed by the reply, so the echo is dropped from the live stream only and never from the stored conversation.

Verified it fails without the fix (`["/skills"]` comes back as a user chunk) and passes with it. Dropped the two unit tests that only exercised the id comparison; kept the ones covering the sanitized-text and id-backfill edge cases, since those document why the match is by id rather than by text.

Comment (yixuanzhong) [reply crates/goose/src/acp/server.rs]: Fair point. Replaced them with an end-to-end test in `acp_server_test.rs` (`test_slash_command_does_not_echo_the_prompt_to_the_client`, 9401f6456) that drives `/skills` through the real ACP server and asserts the observable behaviour:

- the turn streams no `user_message_chunk` back to the client (the client already rendered the prompt),
- the command's reply still arrives as an `agent_message_chunk`,
- `load_session` replays the prompt exactly once, followed by the reply, so the echo is dropped from the live stream only and never from the stored conversation.

Verified it fails without the fix (`["/skills"]` comes back as a user chunk) and passes with it. Dropped the two unit tests that only exercised the id comparison; kept the ones covering the sanitized-text and id-backfill edge cases, since those document why the match is by id rather than by text.

File: crates/goose/src/acp/server.rs [hunk 2]
```diff
@@ -1364,15 +1375,21 @@ impl GooseAcpAgent {
...
         &self,
         content_item: &MessageContent,
         message: &Message,
         session_id: &SessionId,
         target: &SessionAgentTarget,
         tool_requests: &HashMap<String, ToolRequest>,
+        echoed_prompt_id: Option<&str>,
         cx: &ConnectionTo<Client>,
     ) -> Result<(), agent_client_protocol::Error> {
+        if is_echoed_prompt(message, echoed_prompt_id) {
```
Comment (alexhancock) [inline crates/goose/src/acp/server.rs]: I would replace this with inline logic since it's only called once and drop the tests for the method. It's not much value, and the new test you added in `test_slash_command_does_not_echo_the_prompt_to_the_client` is the more meaningful part!

Also there looks to be a merge conflict to address, then this will be ready to go

Comment (yixuanzhong) [reply crates/goose/src/acp/server.rs]: Both done, thanks.

Inlined the check at the top of `handle_message_content` and dropped the helper along with its unit tests, so the end-to-end test is the only cover now. One piece of the helper I kept: the explicit `echoed_prompt_id.is_some()` check. Without it the `None == None` case matches, and `load_session` passes `None` on the replay path, so every id-less user message would be suppressed.

Rebased onto main and the conflict is gone. The rebase also surfaced a second caller of `handle_message_content` that landed in the meantime, in `acp/server/live_voice.rs`; it passes `None`, so voice turns keep streaming the user message as before.

`cargo test -p goose --test acp_server_test slash_command` passes, and clippy is clean on the touched files.
